public/app/core/components/QueryOperationRow/QueryOperationAction.test.tsx

```diff
@@ -1,24 +1,53 @@
-import { shallow } from 'enzyme';
+import { render, screen, fireEvent } from '@testing-library/react';
 import React from 'react';
 
-import { QueryOperationAction } from './QueryOperationAction';
+import { selectors } from '@grafana/e2e-selectors';
 
-describe('QueryOperationAction', () => {
-  it('renders', () => {
-    expect(() => shallow(<QueryOperationAction title="test" icon="panel-add" onClick={() => {}} />)).not.toThrow();
+import { QueryOperationAction, QueryOperationActionProps } from './QueryOperationAction';
+
+const setup = (propOverrides?: Partial<QueryOperationActionProps>) => {
+  const props: QueryOperationActionProps = {
+    icon: 'panel-add',
+    title: 'test',
+    onClick: jest.fn(),
+    disabled: false,
+  };
+
+  Object.assign(props, propOverrides);
+
+  render(<QueryOperationAction {...props} />);
+};
+
+describe('QueryOperationAction tests', () => {
+  it('should render component', () => {
+    setup();
+
+    expect(
+      screen.getByRole('button', { name: selectors.components.QueryEditorRow.actionButton('test') })
+    ).toBeInTheDocument();
   });
-  describe('when disabled', () => {
-    it('does not call onClick handler', () => {
-      const clickSpy = jest.fn();
-      const wrapper = shallow(<QueryOperationAction icon="panel-add" onClick={clickSpy} title="Test action" />);
-      const actionEl = wrapper.find({ 'aria-label': 'Test action query operation action' });
 
-      expect(actionEl).toHaveLength(1);
-      expect(clickSpy).not.toBeCalled();
+  it('should call on click handler', () => {
+    const clickSpy = jest.fn();
+    setup({ disabled: false, onClick: clickSpy });
+
+    expect(clickSpy).not.toHaveBeenCalled();
+    const queryButton = screen.getByRole('button', { name: selectors.components.QueryEditorRow.actionButton('test') });
+
+    fireEvent.click(queryButton);
```

Comment:
can we use `userEvent` here instead? test needs to be made async then as well
```suggestion
    await userEvent.click(queryButton);
```

Reply:
Sure :)